### PR TITLE
Fixes #26278 - GPG sign client repositories

### PR DESCRIPTION
**Pull request by ekohl**
This enforces that the client repo is GPG signed with the same key as the main foreman repository.

(cherry picked from commit 92da00fdf47d80f2c089207c421708c29b4026a9)

### CODE REVIEW

File: packages/foreman/foreman-release/foreman-release.spec [hunk 1]
```diff
@@ -95,19 +96,24 @@ if [[ '%{release}' != *"develop"* ]];then
   VERSION="%{version}"
   sed "s/nightly/${VERSION%.*}/g" -i %{buildroot}%{repo_dir}/*.repo
   sed "s/gpgcheck=0/gpgcheck=1/g" -i %{buildroot}%{repo_dir}/foreman.repo
-  sed "s/gpgcheck=0/gpgcheck=1/g" -i %{buildroot}%{repo_dir}/foreman-rails.repo
+  sed "s/gpgcheck=0/gpgcheck=1/g" -i %{buildroot}%{repo_dir}/foreman-{rails,client}.repo
```
Comment (evgeni) [inline packages/foreman/foreman-release/foreman-release.spec]: same as in #3862 ;)

Comment (ekohl) [reply packages/foreman/foreman-release/foreman-release.spec]: Updated

**Comment by ekohl** [comment]
The repositories should be signed so installing this RPM should allow actual installation. Will test once it's built.

**Comment by ekohl** [comment]
Updated and manually verified on a fresh CentOS 7 that it checks the GPG key when installing.

**Pull request merged**